robusta_krr/core/integrations/prometheus/metrics/memory.py

```
@@ -71,6 +71,36 @@ def get_query(self, object: K8sObjectData, duration: str, step: str) -> str:
 
 
+def PercentileMemoryLoader(percentile: float) -> type[PrometheusMetric]:
+    """
+    A factory for creating percentile memory usage metric loaders.
+    """
+
+    if not 0 <= percentile <= 100:
+        raise ValueError("percentile must be between 0 and 100")
+
+    class PercentileMemoryLoader(PrometheusMetric):
+        def get_query(self, object: K8sObjectData, duration: str, step: str) -> str:
```

Ruff (0.15.15)

[error] Function argument `object` is shadowing a Python builtin

(A002)